src/shared/services/banners/banners.service.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { Observable } from 'rxjs';
+import { HttpHeaders, HttpParams } from '@angular/common/http';
+import createSpyObj = jasmine.createSpyObj;
+import { BannersService } from './banners.service';
+import { AbstractAppConfig } from '../../../app.config';
+import { HttpService } from '../http';
+import { Banner } from '../../domain';
+
+describe('Banner service', () => {
+
+  const MOCK_BANNER: Banner[] = [{
+    bannerDescription: 'test Banner',
+    bannerUrlText: 'test',
+    bannerUrl: 'banner',
+    bannerViewed: true,
+    bannerEnabled: true
+  }]
+  const JID = ['DIVORCE'];
+  let params: HttpParams;
+
+  const API_URL = 'http://aggregated.ccd.reform';
+  const BANNER_URL = API_URL + '/data/internal/banners/';
+
+  let appConfig: any;
+  let httpService: any;
+
+  let bannerService: BannersService;
+
+  describe('Banners()', () => {
+    const MOCK_BANNER1 = { banners: MOCK_BANNER }
+    beforeEach(() => {
+      params = new HttpParams();
+      httpService = createSpyObj<HttpService>('httpService', ['get']);
+      appConfig = createSpyObj<AbstractAppConfig>('appConfig', ['getApiUrl', 'getBannersUrl', 'getCaseDataUrl']);
+      appConfig.getApiUrl.and.returnValue(API_URL);
+      appConfig.getCaseDataUrl.and.returnValue(BANNER_URL);
+      appConfig.getBannersUrl.and.returnValue(BANNER_URL);
+      bannerService = new BannersService(httpService, appConfig);
+    });
+    describe('getBanners()', () => {
```

Comment:
Can you add tests for when you pass a different jurisdiction? Even if it's to make sure the HTTP call isn't made.

Also, what happens if the subscription fires a second time? Do we still only have one banner shown or do we now have an array of banners?

Finally, a test that passes more than one jurisdiction at a time? What happens under such circumstances when:
1. There is a result for some jurisdictions but not the others.
2. There is a result for all of the jurisdictions.
3. There are no results for any jurisdictions.
4. There are results for none of the requested jurisdictions but some for _other_ jurisdictions.